services/github/github-commit-status.tester.js

```diff
@@ -147,6 +87,6 @@ t.create('commit status - 404 with invalid JSON form github')
   )
```

Comment:
Minor point, but AFAIK, this isn't a realistic test case. The 404 response body for this endpoint should be:

```
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/repos/commits/#compare-two-commits"
}
```

Would it be more realistic to test this response as a 5XX status code?

Reply:
We have live tests above of unknown branch and unknown commit. Should we add one for unknown repo?

This test is to make sure the code in the service that parses the error response works as desired, and doesn't flip out if it gets a 404 that doesn't contain valid JSON.

A 5xx error would get rejected normally (and re-thrown by the service). We could add a test for that if you think it seems helpful.

Reply:
So we're not actually expecting the GH to return a non-JSON response for some 404 case, we just want to use some status code that will mean we attempt to parse an invalid body (and 404 is a number which happens to satisfy that need)?

Reply:
The API provides two kinds of 404 response. I think they all provide JSON responses, though normally we don't parse the JSON from 404 responses.

In this case we need to parse it. That way we can determine whether it's the "no common ancestor…" case, which we report as **no common ancestor**, or any other case, which we report as **repo not found**. There are tests above of each of those cases.

Something we're not expecting per se, but which _could_ happen is that GitHub could send some garbage back with a 404 response. We also don't want to crash when that happens. For the most part, we've been removing `invalidJSON` tests from services, not because we don't care about that case anymore, but because that behavior is provided by BaseJsonService, and that code has its own tests.

Here, I'd argue that, this service is providing its own behavior, so it ought to have it's own test.

We need to use a 404 and not a 5xx, because it's only in the case of a 404 that the service receives a NotFound error and follows the response-parsing code path.

https://github.com/badges/shields/pull/3186/files#diff-84ca388a6016d105807cc195db748b5bR75

Reply:
right, yeah I see what you mean now